tests/models/bridgetower/test_modeling_bridgetower.py

```diff
@@ -506,6 +506,10 @@ def test_model_common_attributes(self):
     def test_inputs_embeds(self):
         pass
 
+    @unittest.skip(reason="""Bridge Tower does not have input/output embeddings. Thus this test is not applicable.""")
+    def test_inputs_embeds_matches_input_ids(self):
+        pass
```

Comment:
Shouldn't this (and most other skips) be caught in the skip conditions of the main test? What's missing?

Reply:
Some models have all the things implemented, but do not use "ipnuts_embeds" any where, even though accept it in forward. I dod not remove unused args like this due to BC but can be cleaned up in another PR if needed

Reply:
IMO -- instead of adding custom test code to handle a silent failure, let's make the properly handle the failure (if the user passes `inputs_embeds` to `forward`, raise a `NotImplementedError`, which can be caught in the test)

You've opened a Pandora's box of work in this PR, but it will be great in the long run! 🤗

Reply:
Hmm, so in cases when the model actually accepts but not uses inputs_embeds, we should remove "get_input_embeddings()" method? Or raise error inside forward after the input_embeds are passed?

I would like to entirely remove inputs_embeds from the forward argument list, but this leads to my question: if we should have the same list of arguments in all forwards in all models, regardless of usage? And will it break anything?

Reply:
> if we should have the same list of arguments in all forwards in all models

Not necessarily e.g. vision models don't accept input_ids. All models which are grouped together should have a common subset of inputs, which enable a full forward pass of the model. Most important is that the inputs have standardised names e.g. we don't have `token_ids` for one model, `input_ids` for another and that their behaviour is consistent e.g. `input_embeds` "means" the same thing across models. 

In the case of text models, I'd say yes, if they accept input_ids, then they should also accept input_embeds, and then throw an error within the forward pass after input_embeds are passes.

Reply:
Btw, now the "test_inputs_embeds" can also be cleaned-up and all model-specific skips removed. I can do it later in another PR :)